x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/mapper/SemanticTextFieldMapper.java

```diff
@@ -119,12 +122,22 @@ public class SemanticTextFieldMapper extends FieldMapper implements InferenceFie
 
     public static final TypeParser PARSER = new TypeParser(
         (n, c) -> new Builder(n, c::bitSetProducer, c.getIndexSettings()),
-        List.of(notInMultiFields(CONTENT_TYPE), notFromDynamicTemplates(CONTENT_TYPE))
+        List.of(validateParserContext(CONTENT_TYPE))
     );
 
+    public static BiConsumer<String, MappingParserContext> validateParserContext(String type) {
+        return (n, c) -> {
+            if (InferenceMetadataFieldsMapper.isEnabled(c.getIndexSettings().getSettings()) == false && c.isWithinMultiField()) {
```

Comment:
`InferenceMetadataFieldsMapper.isEnabled(c.getIndexSettings().getSettings()) == false` is a little opaque to read and not immediately obvious that it has to do with multi fields. Can we make this a bit clearer?